packages/lib-classifier/src/store/SubjectStore/ImageAndTextSubject/ImageAndTextSubject.spec.js

```diff
@@ -0,0 +1,154 @@
+import asyncStates from '@zooniverse/async-states'
+
+import { SubjectFactory, WorkflowFactory } from '@test/factories'
+import mockStore from '@test/mockStore'
+import subjectViewers from '@helpers/subjectViewers'
+
+import ImageAndTextSubject from './ImageAndTextSubject'
+
+describe('Model > ImageAndTextSubject', function () {
+  const subjectSnapshot = SubjectFactory.build({
+    content: 'This is test subject content',
+    contentLoadingState: asyncStates.success,
+    locations: [
+      { 'image/png': 'http://localhost:8080/subjectImage.png' },
+      { 'text/plain': 'http://localhost:8080/subjectContent.txt' }
+    ]
+  })
+
+  const successSubjectSnapshot = SubjectFactory.build({
+    locations: [
+      { 'image/png': 'http://localhost:8080/subjectImage.png' },
+      { 'text/plain': 'http://localhost:8080/success.txt' }
+    ]
+  })
+
+  const invalidLocationsSubjectSnapshot = SubjectFactory.build({
+    locations: [
+      { 'text/plain': 'http://localhost:8080/success.txt' },
+      { 'audio/mpeg': 'http://localhost:8080/example.mp3' }
+    ]
+  })
+
+  const failureSubjectSnapshot = SubjectFactory.build({
+    locations: [
+      { 'image/png': 'http://localhost:8080/subjectImage.png' },
+      { 'text/plain': 'http://localhost:8080/failure.txt' }
+    ]
+  })
+
+  const workflowSnapshot = WorkflowFactory.build()
+  let subject
+
+  before(function () {
+    subject = ImageAndTextSubject.create(subjectSnapshot)
+  })
+
+  it('should exist', function () {
+    expect(ImageAndTextSubject).to.be.ok()
+    expect(ImageAndTextSubject).to.be.an('object')
+  })
+
+  it('should have a `locations` property', function () {
+    expect(subject.locations).to.deep.equal(subjectSnapshot.locations)
+  })
+
+  it('should have two locations', function () {
+    expect(subject.locations).to.have.lengthOf(2)
+  })
+
+  it('should have content as expected', function () {
+    expect(subject.content).to.equal(subjectSnapshot.content)
+  })
+
+  it('should have contentLoadingState as expected', function () {
+    expect(subject.contentLoadingState).to.equal(subjectSnapshot.contentLoadingState)
+  })
+
+  describe('with invalid subject locations', function () {
+    it('should throw an error', function () {
+      expect(() => ImageAndTextSubject.create(invalidLocationsSubjectSnapshot)).to.throw()
+    })
+  })
+
+  describe('with location request response that fails', function () {
+    let subject
+
+    before(async function () {
+      nock('http://localhost:8080')
+        .get('/failure.txt')
+        .reply(404)
+
+      subject = ImageAndTextSubject.create(failureSubjectSnapshot)
+
+      const store = mockStore()
+      store.workflows.setResources([workflowSnapshot])
+      store.workflows.setActive(workflowSnapshot.id)
+      store.subjects.setResources([subject])
+      store.subjects.setActive(subject.id)
+
+      await when(() => subject.contentLoadingState === asyncStates.error)
+    })
+
+    after(function () {
+      nock.cleanAll()
```

Comment:
This won't break anything, but you only need to manually clean nock if you use `.persist()`. Without `persist`, your mocked responses are automatically cleaned after they've been requested.